### DISCUSSION TITLE
How to know whether AnalyticsCollectionEnabled is enabled to not?

### DISCUSSION THREAD

**Discussion by rashidul0405** [invertase/react-native-firebase#5254] [Q&A] [answered]
For GDPR, we need to implement a turn on/off switch to enable/disable logging analytics data. We can easily set it enabled/disabled as follows:
```js
// by default true - this need to changed - based on anayticsCollection enabled or not
const [analytics, setAnalytics] = useState(true)

const toggleAnalytics = async () => {
    await firebase.analytics().setAnalyticsCollectionEnabled(!analytics)
    setAnalytics(!analytics)
}

return (
  <Switch
      trackColor={{ true: theme.palette.primary }}
      style={styles.switch}
      onValueChange={toggleAnalytics}
      value={analytics}
  />
)
```

However, after adjusting the settings, we need to able to know whether anayticsCollection enabled or not to initialize the state(switch value) next time. How can we do that? couldn't find anything on doc/codebase.

**Comment by mikehardy** [comment, marked as answer]
There is no API for that in the corresponding module for firebase-js-sdk (which we pattern off of): https://firebase.google.com/docs/reference/js/firebase.analytics.Analytics#setanalyticscollectionenabled

Which implies that you must keep track of it. AsyncStorage is typically how people solve problems like this where they must keep track of state.

**Reply by rashidul0405** [reply]
Thank you ☺️
